src/lando/utils/github.py

```diff
@@ -44,17 +85,40 @@ def _get_token(repo_owner: str, repo_name: str) -> str | None:
 
         if not app_id or not private_key:
             logger.warning(
-                f"Missing GITHUB_APP_ID or GITHUB_APP_PRIVKEY to authenticate against GitHub repo {repo_owner}/{repo_name}",
+                f"Missing GITHUB_APP_ID or GITHUB_APP_PRIVKEY to authenticate against GitHub repo at {self.repo_url}",
             )
             return None
 
         app_auth = AppAuth(
             app_id,
             private_key,
         )
-        session = AppInstallationAuth(app_auth, repo_owner, repositories=[repo_name])
+        session = AppInstallationAuth(
+            app_auth, self.repo_owner, repositories=[self.repo_name]
+        )
         return asyncio.run(session.get_token())
 
+
+class GitHubAPI(GitHub):
+    """A simple wrapper that authenticates with and communicates with the GitHub API."""
+
+    session: requests.Session
+
+    GITHUB_BASE_URL = "https://api.github.com"
+
+    def __init__(self, repo_url: str):
+        super().__init__(repo_url)
+
+        self.session = requests.Session()
+        self.session.headers.update(
+            {
+                "Authorization": f"Bearer {self._fetch_token()}",
+                "User-Agent": settings.HTTP_USER_AGENT,
+                "Accept": "application/vnd.github+json",
+                "X-GitHub-Api-Version": "2022-11-28",
+            }
+        )
```

Comment:
Make sure to add a comment in the commit message about adding session functionality, since this is out of scope of the original PR. I would also suggest adding a bug reference to this PR at this point because this will go directly into #631 and will end up with its own commit in `main`..